agent/session.go

```diff
@@ -402,6 +405,9 @@ func (s *session) sendError(err error) {
 // of event loop.
 func (s *session) close() error {
 	s.closeOnce.Do(func() {
+		if s.cancel != nil {
```

Comment:
I don't believe it is possible for `s.cancel` to be nil. It is filled in when the session is created in `newSession`.